junit-jupiter-migration-support/src/main/java/org/junit/jupiter/migrationsupport/rules/ParameterizedExtension.java

```diff
@@ -0,0 +1,294 @@
+/*
+ * Copyright 2015-2017 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v1.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.junit.jupiter.migrationsupport.rules;
+
+import static java.util.Collections.singletonList;
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.toList;
+import static org.junit.platform.commons.meta.API.Usage.Experimental;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.extension.BeforeTestExecutionCallback;
+import org.junit.jupiter.api.extension.ContainerExtensionContext;
+import org.junit.jupiter.api.extension.Extension;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+import org.junit.jupiter.api.extension.TestExtensionContext;
+import org.junit.jupiter.api.extension.TestTemplateInvocationContext;
+import org.junit.jupiter.api.extension.TestTemplateInvocationContextProvider;
+import org.junit.platform.commons.meta.API;
+import org.junit.platform.commons.util.ReflectionUtils;
+import org.junit.runners.Parameterized;
+
+@API(Experimental)
+public class ParameterizedExtension implements TestTemplateInvocationContextProvider, ParameterResolver {
+	private static ExtensionContext.Namespace parameters = ExtensionContext.Namespace.create(
+		ParameterizedExtension.class);
+	private int parametersCollectionIndex = 0;
+
+	/**
+	 * Indicate whether we can provide parameterized support.
+	 * This requires the testClass to either have a static {@code @Parameters} method
+	 * and correct {@code @Parameter} and their corresponding values
+	 * or to have a constructor that could be injected.
+	 */
+	@Override
+	public boolean supports(ContainerExtensionContext context) {
+		return hasParametersMethod(context) && hasCorrectParameterFields(context);
+	}
+
+	@Override
+	public Iterator<TestTemplateInvocationContext> provide(ContainerExtensionContext context) {
```

Comment:
@marcphilipp no problem at all, the iterator was being created from a stream anyways. So it actually simplified the code!